### ISSUE TITLE
ExodusIIDataset._is_valid doesn't clean up file handles properly

### ISSUE THREAD

**Issue by ngoldbaum** [yt-project/yt#1652]
This came up in the context of #1645. When running the following script:

```
from yt.frontends.exodus_ii.data_structures import ExodusIIDataset
from yt.frontends.ytdata.data_structures import YTDataLightRayDataset
ExodusIIDataset._is_valid('singleParticle/simData.h5')
YTDataLightRayDataset._is_valid('singleParticle/simData.h5')
```

I get the following traceback (but only on the machine that runs the yt test suite):

```
File "test.py", line 5, in <module>
    YTDataLightRayDataset._is_valid('/mnt/yt/singleParticle/simData.h5')
  File "/home/fido/workspace/yt_py3_git/yt/frontends/ytdata/data_structures.py", line 336, in _is_valid
    with h5py.File(args[0], "r") as f:
  File "/usr/lib64/python3.5/site-packages/h5py/_hl/files.py", line 271, in __init__
    fid = make_fid(name, mode, userblock_size, fapl, swmr=swmr)
  File "/usr/lib64/python3.5/site-packages/h5py/_hl/files.py", line 101, in make_fid
    fid = h5f.open(name, flags, fapl=fapl)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper (/var/tmp/portage/dev-python/h5py-2.7.0/work/h5py-2.7.0/h5py/_objects.c:2840)
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper (/var/tmp/portage/dev-python/h5py-2.7.0/work/h5py-2.7.0/h5py/_objects.c:2798)
  File "h5py/h5f.pyx", line 78, in h5py.h5f.open (/var/tmp/portage/dev-python/h5py-2.7.0/work/h5py-2.7.0/h5py/h5f.c:2117)
OSError: Unable to open file (File close degree doesn't match)
```

**Comment by ngoldbaum** [comment]
Oh wait, sorry, false alarm. I misread the error message. Let's see if I can reproduce this for real...

**Comment by ngoldbaum** [comment]
OK, was able to reproduce it (but only on fido) :(

**Comment by MatthewTurk** [comment]
Evidently this is fixed in #1868 ?
